src/main/java/dansplugins/factionsystem/commands/abs/TabCompleterBase.java

```diff
@@ -0,0 +1,588 @@
+package dansplugins.factionsystem.commands.abs;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import dansplugins.factionsystem.data.PersistentData;
+import dansplugins.factionsystem.objects.domain.Faction;
+import dansplugins.factionsystem.services.ConfigService;
+import org.bukkit.Bukkit;
+import org.bukkit.OfflinePlayer;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+import org.bukkit.command.TabCompleter;
+import org.bukkit.entity.Player;
+
+import javax.annotation.Nullable;
+
+public class TabCompleterBase implements TabCompleter {
+
+	PersistentData persistentData;
+	ConfigService configService;
+	public TabCompleterBase(PersistentData persistentData, ConfigService configService)
+	{
```

Comment:
It would be good to maintain a consistent bracketing style with the rest of the project

Reply:
That would be preferable, yeah. We haven't actually defined the official bracketing style so this is something we should handle another time IMO.

Reply:
I've modified the bracketing style in this file.